src/mito2/src/manifest/storage.rs

```diff
@@ -380,7 +404,7 @@ impl ManifestObjectStore {
         let checkpoint_metadata = CheckpointMetadata {
             size: bytes.len(),
             version,
-            checksum: None,
+            checksum: Some(checksum),
```

Comment:
I'm wondering if we can use `u32` here. Previous `null` can be decoded to default `0` and is always mismatched?

Reply:
If so, it will break the compatibility?

Reply:
You're right that we should then skip checksum verification on `0`, which is effectively the same as skip for `None`.

Keep the current impl.